src/ServiceControl.Plugin.Nsb6.SagaAudit/SagaAudit.cs

```diff
@@ -13,13 +15,26 @@ public SagaAudit()
             EnableByDefault();
             DependsOn<Sagas>();
         }
-        
+
         protected override void Setup(FeatureConfigurationContext context)
         {
             context.Container.ConfigureComponent<ServiceControlBackend>(DependencyLifecycle.SingleInstance);
 
             context.Pipeline.Register<CaptureSagaStateRegistration>();
             context.Pipeline.Register<CaptureSagaResultingMessageRegistration>();
+
+            context.Container.RegisterSingleton(BuildSerializer(context.Settings));
+        }
+
+        static CaptureSagaStateSerializer BuildSerializer(ReadOnlySettings settings)
+        {
+            var definition = new JsonSerializer();
+
+            var factory = definition.Configure(settings);
+
+            var serializer = factory(new MessageMapper());
+
+            return new CaptureSagaStateSerializer(serializer);
```

Comment:
The idea is to initialize our `JsonMessageSerializer` in the same way we do in the core by using the `JsonSerializer`. Then wrap it in a new class with the old [Serializer](https://github.com/Particular/ServiceControl.Plugin.Nsb5.SagaAudit/blob/master/src/ServiceControl.Plugin.Nsb5.SagaAudit/Serializer.cs) implementation and register it so it can be injected into the behavior.